### PR TITLE
*: fix structcheck lint warnings

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve?

Issue Number: https://github.com/pingcap/tidb/issues/22979

### What is changed and how it works?

What's Changed:
- remove some unused struct field
- disable linter for some false positive cases
- update makefile to add linter

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -427,10 +427,10 @@ type ShowDDLJobsExec struct {
 
 	jobNumber int
 	is        infoschema.InfoSchema
-	done      bool
 }
 
 // DDLJobRetriever retrieve the DDLJobs.
+// nolint:structcheck
 type DDLJobRetriever struct {
```
Comment (zimulala) [inline executor/executor.go:434]: Why this struct need "nolint..."?

Comment (disksing) [reply executor/executor.go:434]: It has fields that it does not access by itself but used by other structs that embed it. Such as `ShowDDLJobsExec`. I think it is a false positive report.

Comment (zimulala) [reply executor/executor.go:434]: Got it.

**Review by Yisaer — APPROVED** [review]
LGTM

**Review by JmPotato — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**